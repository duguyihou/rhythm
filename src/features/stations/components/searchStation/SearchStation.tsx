import { ChangeEvent, KeyboardEvent, useState } from 'react'

import { Dialog } from '../../../../components/elements'
import { useDisclosure } from '../../../../hooks/useDisclosure'
import { useStationSearchByName } from '../../hooks'

const SearchStation = () => {
  const [input, setInput] = useState('')
  const { data, refetch } = useStationSearchByName({ name: input })
  const { close, open, isOpen } = useDisclosure()

  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
    setInput(event.target.value)
  }
  const handleOnKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      event.preventDefault()
      input !== '' && refetch()
    }
  }

  return (
    <div className="flex flex-col">
      <div className="flex flex-row items-center rounded-sm relative">
        <input
          value={input}
          className="p-1 m-4 flex-1 focus:outline-0 rounded-md"
          type="text"
          placeholder="Search..."
          onChange={handleOnChange}
          onKeyDown={handleOnKeyDown}
          onFocus={open}
          onBlur={close}
        />
        <div className="absolute right-4">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="22"
            height="20"
            aria-hidden="true"
            className="mr-1 header-search-key-slash"
          >
            <path
              fill="none"
              stroke="#979A9C"
              opacity=".4"
              d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z"
            ></path>
            <path fill="#979A9C" d="M11.8 6L8 15.1h-.9L10.8 6h1z"></path>
          </svg>
        </div>
      </div>
      <Dialog isOpen={isOpen} onClose={close}>
        {data?.map((station) => (
          <div className="p-2" key={station.stationuuid}>
            {station.name}
          </div>
        ))}
      </Dialog>
    </div>
  )
}

export default SearchStation
