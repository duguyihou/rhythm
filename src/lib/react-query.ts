/* eslint-disable @typescript-eslint/no-explicit-any */
import { QueryClient } from '@tanstack/react-query'
import type {
  DefaultOptions,
  UseMutationOptions,
  UseQueryOptions,
} from '@tanstack/react-query'

const queryConfig: DefaultOptions = {
  queries: {
    useErrorBoundary: true,
    refetchOnWindowFocus: false,
    retry: false,
  },
}

export const queryClient = new QueryClient({ defaultOptions: queryConfig })

export type ExtractFnReturnType<FnType extends (...args: any) => any> = Promise<
  ReturnType<FnType>
>

export type QueryConfig<QueryFnType extends (...args: any) => any> = Omit<
  UseQueryOptions<ExtractFnReturnType<QueryFnType>>,
  'queryKey' | 'queryFn'
>

export type MutationConfig<MutationFnType extends (...args: any) => any> =
  UseMutationOptions<
    ExtractFnReturnType<MutationFnType>,
    Error,
    Parameters<MutationFnType>[0]
  >
