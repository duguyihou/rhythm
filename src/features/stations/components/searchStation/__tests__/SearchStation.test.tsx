// eslint-disable-next-line import/named
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'

import SearchStation from '../SearchStation'

test('searchStation', () => {
  const queryClient = new QueryClient()
  const searchStation = render(
    <QueryClientProvider client={queryClient}>
      <SearchStation />
    </QueryClientProvider>
  )

  expect(searchStation).toMatchSnapshot()
})
