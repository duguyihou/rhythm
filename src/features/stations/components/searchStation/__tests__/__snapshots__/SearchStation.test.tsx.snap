// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchStation 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row items-center rounded-sm relative"
        >
          <input
            class="p-1 m-4 flex-1 focus:outline-0 rounded-md"
            placeholder="Search..."
            type="text"
            value=""
          />
          <div
            class="absolute right-4"
          >
            <svg
              aria-hidden="true"
              class="mr-1 header-search-key-slash"
              height="20"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z"
                fill="none"
                opacity=".4"
                stroke="#979A9C"
              />
              <path
                d="M11.8 6L8 15.1h-.9L10.8 6h1z"
                fill="#979A9C"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-row items-center rounded-sm relative"
      >
        <input
          class="p-1 m-4 flex-1 focus:outline-0 rounded-md"
          placeholder="Search..."
          type="text"
          value=""
        />
        <div
          class="absolute right-4"
        >
          <svg
            aria-hidden="true"
            class="mr-1 header-search-key-slash"
            height="20"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z"
              fill="none"
              opacity=".4"
              stroke="#979A9C"
            />
            <path
              d="M11.8 6L8 15.1h-.9L10.8 6h1z"
              fill="#979A9C"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
